src/Raven.Server/Documents/Archive/ArchivalStorage.cs

```diff
@@ -0,0 +1,341 @@
+                        builder.WriteValue((LazyStringValue)counterNamesDict[counterNameKey]);
+                    }
+                }
+                
+                builder.WriteObjectEnd();
+                
+                builder.WriteObjectEnd();
+                builder.FinalizeDocument();
+
+                mergedCounterGroupValues = builder.CreateReader();
+            }
+
+            return mergedCounterGroupValues;
+        }
+
+        
+        private BlittableJsonReaderObject CollectArchivedDocumentCountersIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+            if ((doc.Flags & DocumentFlags.HasCounters) == 0) return null;
+            
+            var counterGroups = _database.DocumentsStorage.CountersStorage.GetCounterValuesForDocument(context, documentId).ToList();
+                                            
+            var clonedCountersGroupsValues = new List<BlittableJsonReaderObject>(counterGroups.Count);
+            foreach (var counterGroup in counterGroups)
+                clonedCountersGroupsValues.Add(counterGroup.Values.Clone(context));
+                                            
+            var mergedCounterGroupsValues = BuildMergedCounterGroupValues(context, clonedCountersGroupsValues, documentId);
+                                                
+            _database.DocumentsStorage.CountersStorage.DeleteCountersForDocument(context, documentId, new CollectionName(collectionName));
+
+            return mergedCounterGroupsValues;
+        }
+
+        private void PutArchivedCounters(DocumentsOperationContext context, string documentId, string archivedCollectionName, BlittableJsonReaderObject mergedCounterGroupsValues)
+        {
+            var newCounterEtag = _database.DocumentsStorage.GenerateNextEtag();
+            var newCounterChangeVector = _database.DocumentsStorage.GetNewChangeVector(context, newCounterEtag);
+            _database.DocumentsStorage.CountersStorage.PutCounters(context, documentId, archivedCollectionName, newCounterChangeVector , mergedCounterGroupsValues, newCounterEtag);
+        }
+
+        private void ArchiveDocumentTimeSeriesIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+            
+        }
+        
+        private void ArchiveDocumentRevisionsIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+            
+        }
+        
+        private void ArchiveDocumentAttachmentsIfNeeded(DocumentsOperationContext context, Document doc, string documentId, string collectionName, string archivedCollectionName)
+        {
+                    
+        }
+
+        private bool MigrateDocument(DocumentsOperationContext context, Slice lowerId, string id, DateTime currentTime)
```

Comment:
Migrate? Wouldn't "archive" be more suitable?

Reply:
https://github.com/ravendb/ravendb/pull/15715/commits/9121b8dc677788428f888e5ada41ff3ae6186cc2 fixed